docs/circuit_overview.md

```diff
@@ -0,0 +1,267 @@
+## Airbender Circuit Overview
+
+This document provides a high level overview of Airbender proving system for RISC‑V execution and then dives into key design patterns and stages. If you just want the big picture, read this section. Later sections explain how each part is implemented and optimized.
```

Comment:
Is it an overview of the proof system or just the circuits? I think the former, so the title of the document is misleading